urlcdab7d8364_jrwalsh_CycTools_tgz-pJ8-edu_iastate_cyctools_MainJ8/src/edu/iastate/cyctools/tools/load/view/LoadPanel.java

```diff
@@ -0,0 +1,1359 @@
+import javax.swing.text.BadLocationException;
+import javax.swing.text.DefaultHighlighter;
+import javax.swing.text.Highlighter.Highlight;
+import javax.swing.AbstractAction;
+import javax.swing.Action;
+import javax.swing.DefaultComboBoxModel;
+import javax.swing.DefaultListCellRenderer;
+import javax.swing.DefaultListModel;
+import javax.swing.JButton;
+import javax.swing.JFileChooser;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JTable;
+
+import java.awt.event.ActionEvent;
+import java.awt.Color;
+import java.awt.Component;
+import java.awt.Dimension;
+import java.awt.Font;
+import java.beans.PropertyChangeEvent;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.Map.Entry;
+
+import edu.iastate.cyctools.CycToolsError;
+import edu.iastate.cyctools.DefaultController;
+import edu.iastate.cyctools.InternalStateModel.State;
+import edu.iastate.cyctools.externalSourceCode.AbstractViewPanel;
+import edu.iastate.cyctools.externalSourceCode.DiffMatchPatch.Patch;
+import edu.iastate.cyctools.externalSourceCode.KeyValueComboboxModel;
+import edu.iastate.cyctools.externalSourceCode.DiffMatchPatch;
+import edu.iastate.cyctools.externalSourceCode.DiffMatchPatch.Diff;
+import edu.iastate.cyctools.tools.load.fileAdaptors.AbstractFileAdaptor;
+import edu.iastate.cyctools.tools.load.fileAdaptors.CreateRegulationAdaptor;
+import edu.iastate.cyctools.tools.load.fileAdaptors.DeleteFrameAdaptor;
+import edu.iastate.cyctools.tools.load.fileAdaptors.GOTermAdaptor;
+import edu.iastate.cyctools.tools.load.fileAdaptors.SimpleAnnotationValueImport;
+import edu.iastate.cyctools.tools.load.fileAdaptors.SimpleSlotValueImport;
+import edu.iastate.cyctools.tools.load.model.BatchUpdate.Event;
+import edu.iastate.cyctools.tools.load.model.BatchUpdate.Status;
+import edu.iastate.cyctools.tools.load.model.DocumentModel;
+import edu.iastate.cyctools.tools.load.model.BatchUpdate;
+import edu.iastate.javacyco.Frame;
+import edu.iastate.javacyco.JavacycConnection;
+import edu.iastate.javacyco.PtoolsErrorException;
+
+import java.awt.CardLayout;
+import javax.swing.JComboBox;
+import javax.swing.JCheckBox;
+import javax.swing.JTextField;
+import javax.swing.JLabel;
+import javax.swing.JList;
+import javax.swing.JTextArea;
+import javax.swing.ListSelectionModel;
+import javax.swing.event.ListSelectionListener;
+import javax.swing.event.ListSelectionEvent;
+import javax.swing.JTextPane;
+import javax.swing.JTabbedPane;
+import javax.swing.ImageIcon;
+import javax.swing.event.ChangeListener;
+import javax.swing.event.ChangeEvent;
+import net.miginfocom.swing.MigLayout;
+import java.awt.event.ActionListener;
+import javax.swing.border.EtchedBorder;
+
+@SuppressWarnings({"serial", "rawtypes", "unchecked"})
+public class LoadPanel extends AbstractViewPanel {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This class references 84 other classes. By comparison, 99% of the classes in the CodeGuru reference dataset reference fewer. This indicates that this class is highly coupled with other classes. A class that is highly coupled with other classes is difficult to understand and its behavior might change unexpectedly when one of its referenced classes is updated. High coupling could also increase the integration test complexity, maintenance cost and technical debt. We recommend that you simplify this class or break it into multiple classes.